### PR TITLE
Add clang format ci job

### LINKED ISSUE #887: Add CI job that checks formatting with clang-format
## Brief feature description

iceoryx uses clang-format to format files and it will be helpful to have this as part of CI

## Detailed information

N/A

### PR DESCRIPTION
**Pull request by r7vme**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CHANGELOG.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #887

### CODE REVIEW

File: tools/check_formatting.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+CLANG_MAJOR_VERSION=$(clang-format --version | awk '{print $(NF)}' | sed 's/\..*$//')
+[[ $CLANG_MAJOR_VERSION -ge "10" ]] || fail "clang-format of version 10 or higher is not installed."
+
+cd $(git rev-parse --show-toplevel)
+
+# format files
+git ls-files | grep -E "\.(c|cpp|inl|h|hpp)$" | xargs clang-format -i -style=file
+
+# check incorrectly formatted fails and fail
+git diff-index --name-only --exit-code HEAD --
```
Comment (r7vme) [inline tools/check_formatting.sh:20]: this will return 1 and fail scrip if there are non-commited changes (produced by clang-format)

**Comment by r7vme** [comment]
@dkroenke @elBoberido PTAL

**Review by elBoberido — DISMISSED** [review]

File: tools/check_formatting.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (elBoberido) [inline tools/check_formatting.sh:1]: please add the copyright and license header like in the other shell scripts

Comment (r7vme) [reply tools/check_formatting.sh:1]: done

File: tools/check_formatting.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+set -e
+
+fail() {
+    printf "\033[1;31merror: %s: %s\033[0m\n" ${FUNCNAME[1]} "${1:-"Unknown error"}"
+    exit 1
+}
+
+hash git || fail "git not found"
+hash clang-format-10 || fail "clang-format-10 not found"
```
Comment (r7vme) [inline tools/check_formatting.sh:30]: we agreed with @dkroenke that CI scripts should stick with ONE version of tool, which will make this more deterministic and require less maintenance in the future

**Review by elBoberido — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

**Pull request merged**